pkg/aws/core.go

```diff
@@ -368,7 +369,7 @@ func (d *Driver) DeleteMachine(_ context.Context, req *driver.DeleteMachineReque
 
 	} else {
 		// ProviderID doesn't exist, hence check for any existing machine and then delete if exists
-		instances, err = d.getInstancesFromMachineName(req.Machine.Name, providerSpec, req.Secret)
+		instances, err = d.getMatchingInstancesForMachine(req.Machine, providerSpec, req.Secret)
```

Comment:
In the `If` condition you have already checked that the providerID is empty. Therefore just calling `getMachineInstancesByTagsAndStatus` should be sufficient, right? This also prevents recreating the service which you have already created once.